python/pyiceberg/catalog/rest.py

```diff
@@ -335,6 +349,53 @@ def _handle_non_200_response(self, exc: HTTPError, error_handler: Dict[int, Type
 
         raise exception(response) from exc
 
+    def _init_sigv4(self) -> None:
+        from urllib import parse
+
+        import boto3
+        from botocore.auth import SigV4Auth
+        from botocore.awsrequest import AWSRequest
+        from requests import PreparedRequest
+        from requests.adapters import HTTPAdapter
+
+        class SigV4Adapter(HTTPAdapter):
+            def __init__(self, **properties: str):
+                super().__init__()
+                self._properties = properties
+
+            def add_headers(self, request: PreparedRequest, **kwargs: Any) -> None:  # pylint: disable=W0613
+                session = boto3.Session()
+                credentials = session.get_credentials()
+                creds = credentials.get_frozen_credentials()
```

Comment:
nit. Maybe do:

```python
credentials = session.get_credentials().get_frozen_credentials()
```

This way we avoid similar variable names `creds` and `credentials`